pydatastructs/linear_data_structures/algorithms.py

```diff
@@ -722,3 +723,67 @@ def partition(low, high, pick_pivot_element):
         array._modify(force=True)
 
     return array
+
+def longest_common_subsequence(seq1, seq2) -> tuple:
+    """
+    Implements Longest Common Subsequence
+
+    Parameters
+    ========
+
+    seq1: String or List or Tuple
+    seq2: String or List or Tuple
+
+    Returns
+    =======
+
+    output: tuple
+    (Length of LCS, Common Sequence)
+    Common Sequence will be of the same data type as seq1.
+
+    Examples
+    ========
+
+    >>> from pydatastructs import longest_common_subsequence as LCS
+    >>> LCS("ABCDEF", "ABBCDDDE")
+    (5, 'ABCDE')
+    >>> arr1 = ['A', 'P', 'P']
+    >>> arr2 = ['A', 'p', 'P', 'S', 'P']
+    >>> LCS(arr1, arr2)
+    (3, ['A', 'P', 'P'])
+
+    References
+    ==========
+
+    .. [1] https://en.wikipedia.org/wiki/Longest_common_subsequence_problem
+    """
+    if not isinstance(seq1, (str, tuple, list)):
+        raise TypeError("Only Strings, Tuple and List are allowed")
+    if not isinstance(seq2, (str, tuple, list)):
+        raise TypeError("Only Strings, Tuple and List are allowed")
+
+    row, col = len(seq1), len(seq2)
+    check_mat = [[0 for _ in range(col+1)] for x in range(row+1)]
```

Comment:
What about using a nested `dict`. AFAICT, half of the matrix in these type of problems is untouched and wasted. `dict` will be at least theoretically better.